starknet_devnet/starknet_wrapper.py

```diff
@@ -317,18 +310,24 @@ async def __aexit__(
 
                 if exc_type:
                     assert isinstance(exc, StarkException)
-                    error_message = exc.message
                     status = TransactionStatus.REJECTED
-                    self.execution_info = TransactionExecutionInfo.empty()
-                    state_update = None
 
                     # restore block info
                     self.starknet_wrapper.get_state().state.block_info = (
                         self.preserved_block_info
                     )
+
+                    transaction = DevnetTransaction(
+                        internal_tx=self.internal_tx,
+                        status=status,
+                        execution_info=self.execution_info,
+                        transaction_hash=tx_hash,
+                    )
+                    self.starknet_wrapper._store_transaction(
+                        transaction, error_message=exc.message
+                    )
                 else:
-                    error_message = None
-                    status = TransactionStatus.ACCEPTED_ON_L2
+                    status = TransactionStatus.PENDING
```

Comment:
Will it be a breaking change for our users? If yes, will it be documented enough?